### ISSUE TITLE
WebAPI JUnit tests need documentation and update

### ISSUE THREAD

**Issue by chrisknoll** [OHDSI/WebAPI#239]
I've spent some time trying to get the IT (integration tests) in WebAPI to run and I'm not sure if it's an issue with the updates to the spring boot library, but the tests are not completely functional, from what I can tell.

By default we have the pom.xml specify to skip tests:

```
<plugin>
        <groupId>org.apache.maven.plugins</groupId>
        <artifactId>maven-surefire-plugin</artifactId>
        <version>2.18.1</version>
        <configuration>
          <skipTests>true</skipTests>
        </configuration>
</plugin>
```

So we haven't been running tests in a while.

I think we need to revisit the way tests are being coded, or maybe all that's required is documentation, but it's not clear about how tests should be executed and implemented.

**Comment by t-abdul-basser** [comment]
I agree fully. Thank you reviving this discussion @chrisknoll . If I get time I will make a unit and integration testing policy and framework proposal in the near future.

**Comment by leeevans** [comment]
@chrisknoll and @t-abdul-basser I'm interested in this topic to.  

It would be good to run tests in the automated Jenkins builds. I believe there was a slow AWS RDS Postgres database issue that was a blocker in the past but I could investigate that with the latest code. 

How about also running a nightly Jenkins build job with tests against the different DBMS platforms - Redshift, Postgres, SQL Server, Oracle (and maybe Impala & BigQuery) so we can maintain ongoing compatibility with all the supported DBMSs? Maybe initially focused on Achilles?
If we could get some requirements together I could help set that up.

**Comment by anthonysena** [comment]
Leaving this open as it relates to #2243
